### PR TITLE
benchmark evals + api routes

### PR DESCRIPTION
**Pull request by bboinay**
please squash this whole branch! Lot's of tiny debug commits, none of which need to be kept in history. If you're unsure let's pair up before merging.

### CODE REVIEW

File: eval_pg_utils.py [hunk 1]
```diff
@@ -68,22 +75,32 @@ def import_csv_to_golden_dataset(csv_file_path):
...
         
+        # Clear the existing data from the benchmark_data table
+        clear_query = "DELETE FROM benchmark_data"
+        cursor.execute(clear_query)
+        print("Existing data in 'benchmark_data' table has been cleared.")
+        log.info("Existing data in 'benchmark_data' table has been cleared.")
+
         # Prepare the SQL query
-        insert_query = f"INSERT INTO golden_dataset (data) VALUES (%s)"
+        insert_query = f"INSERT INTO benchmark_data (data) VALUES (%s)"
```
Comment (yg-lim) [inline eval_pg_utils.py:85]: does this string interpolation in Python allow for SQL injection?

File: server.py [hunk 2]
```diff
@@ -56,39 +66,88 @@ async def upload(file: UploadFile=File(...)):
...
 
-@app.post('/api/csv')
+# simple pipeline query with no side effects
+# TESTED, WORKING 7/29/24
+# required body for Postman
+    # body -> raw -> json
+    # {
+    #     "query": "how tall are banana trees"
+    # }
+@app.post('/api/query')
```
Comment (yg-lim) [inline server.py:76]: want to know more about this route after we get the feature validation screenshots!

**Review by yg-lim — COMMENTED** [review]
had a Q about SQL injection and waiting on the feature validation stuff. looks good otherwise!

File: eval_pg_utils.py (hunk 1 above)
Comment (bboinay) [reply eval_pg_utils.py]: Great call; it indeed does. Fixed!

File: server.py [hunk 3]
```diff
@@ -24,18 +24,28 @@
...
+
+# QUERY/CHAT ROUTES
+
+# api route for document uploads to add to document collection
+# TESTED, WORKING 7/29/24
+# required key/values for Postman:
+    # file => upload file via Postman
+    # filename => file name
+@app.post('/api/upload') 
 async def upload(file: UploadFile=File(...)):
```
Comment (jjmchew) [inline server.py:49]: This function name is in conflict (will be shadowed) by the later file upload route - we may need to better organize route naming. Perhaps we could put all eval-specific routes under /api/evals

**Review by jjmchew — CHANGES_REQUESTED** [review]
I left a comment about route naming - should we fix this before we merge?

File: server.py (hunk 3 above)
Comment (bboinay) [reply server.py:49]: renamed '/api/benchmark/upload' function to 'benchmark_upload'

**Review by jjmchew — APPROVED** [review]
lgtm

**Pull request merged**